sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,494 @@
+                    self._internal_forces.pop(label)
+                    self._members[self._members.index([label, member[1], member[2]])] = [new_label, member[1], member[2]]
+
+    def apply_load(self, location, magnitude, direction):
+        """
+        This method applies an external load at a particular node
+
+        Parameters
+        ==========
+        location: String or Symbol
+            Label of the Node at which load is applied.
+
+        magnitude: Sympifyable
+            Magnitude of the load applied. It must always be positive and any changes in
+            the direction of the load are not reflected here.
+
+        direction: Sympifyable
+            The angle, in degrees, that the load vector makes with the horizontal
+            in the counter-clockwise direction. It takes the values 0 to 360,
+            inclusive.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> from sympy import symbols
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> P = symbols('P')
+        >>> t.apply_load('A', P, 90)
+        >>> t.apply_load('A', P/2, 45)
+        >>> t.apply_load('A', P/4, 90)
+        >>> t.loads
+        {'A': [[5*P/4, 90], [P/2, 45]], 'B': [[0, 90]]}
+        """
+        magnitude = sympify(magnitude)
+        direction = sympify(direction)
+
+        if location not in self.node_labels:
+            raise ValueError("Load must be applied at a known node")
+
+        else:
+            if direction in [self._loads[location][i][1] for i in range(len(self._loads[location]))]:
+                for i in range(len(self._loads[location])):
+                    if self._loads[location][i][1] == direction:
+                        self._loads[location][i][0] += magnitude
+                        break
+            else:
+                self._loads[location].append([magnitude, direction])
+
+    def add_support(self, location, type):
+        """
+        This method adds a pinned or roller support at a particular node
+
+        Parameters
+        ==========
+
+        location: String or Symbol
+            Label of the Node at which support is added.
+
+        type: String
+            Type of the support being provided at the node.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.add_support('A', 'pinned')
+        >>> t.supports
+        {'A': 'pinned', 'B': 'none'}
+        """
+        if location not in self._node_labels:
+            raise ValueError("Support must be added on a known node")
+
+        else:
+            self._supports[location] = type
+
+        if type == "pinned":
+            self._reaction_forces['R_'+str(location)+'_x'] = 0
+            self._reaction_forces['R_'+str(location)+'_y'] = 0
```

Comment:
Instead of adding reaction forces to `self._reaction_forces` and setting them to 0, you should apply a load at the node with magnitude = symbol('R_'+str(location)+'_x') for respective directions. The forces should be added to `self._reaction_forces` only after solve function is called, which will calculate the actual magnitude.

Calling the function `self.reaction_forces()` should return an empty list before the solve function is called.
`self._reaction_forces` should contain only the solved reaction loads
See how the beam module handles this [here](https://github.com/sympy/sympy/blob/ce53e7b5aeee7019cf019b9e7cbea07fe3f8cbac/sympy/physics/continuum_mechanics/beam.py#L419)